### DISCUSSION TITLE
umijs3.5中使用web worker

### DISCUSSION THREAD

**Discussion by JasonHuo-J** [umijs/umi#12026] [Q&A]
我在umi中使用web worker时遇到了一些问题，我已经在config中配置了webpack5:{}，
主进程中
`useEffect(() => {
    if (typeof Worker !== 'undefined') {
      const myworker = new Worker(new URL('./worker.worker.js', import.meta.url));
      myworker.onmessage = (e: any) => {
        console.log(e.data);
      };
      myworker.onerror = (e) => {
        console.log(e);
      };
      myworker.postMessage('index.ts');
    }
  }, []);`
  
  worker进程
  `self.onmessage = (e) => {
  console.log('worker.ts', e);
  postMessage('worker.ts');
};
`
然后就出现了错误
![image](https://github.com/umijs/umi/assets/56123635/feb2b01d-d54b-4a2c-a520-62d3f1330ef0)

请问是什么原因？

**Comment by fz6m** [comment]
umi 3 不维护了，建议升级到 umi 4
